src/devices_models/devices/DC_branches.jl

```diff
@@ -76,52 +77,52 @@ function branch_rate_constraints!(
     return
 end
 
-branch_rate_constraints!(
+add_constraints!(
     ::OptimizationContainer,
+    ::Type{<:Union{FlowRateConstraintFromTo, FlowRateConstraintToFrom, FlowRateConstraint}},
     ::IS.FlattenIteratorWrapper{<:PSY.DCBranch},
     ::DeviceModel{<:PSY.DCBranch, HVDCUnbounded},
     ::Type{<:PM.AbstractPowerModel},
     ::Union{Nothing, AbstractAffectFeedForward},
 ) = nothing
 
-function branch_rate_constraints!(
+function add_constraints!(
     container::OptimizationContainer,
+    cons_type::Type{
+        <:Union{FlowRateConstraintFromTo, FlowRateConstraintToFrom, FlowRateConstraint},
+    },
     devices::IS.FlattenIteratorWrapper{B},
     ::DeviceModel{B, <:AbstractDCLineFormulation},
     ::Type{<:PM.AbstractPowerModel},
     feedforward::Union{Nothing, AbstractAffectFeedForward},
 ) where {B <: PSY.DCBranch}
     time_steps = get_time_steps(container)
     names = [PSY.get_name(d) for d in devices]
-    for (var_type, cons_type) in zip(
-        (FlowActivePowerVariable(), FlowActivePowerVariable()),
-        (FlowRateConstraintFT(), FlowRateConstraintTF()),
-    )
-        var = get_variable(container, var_type, B)
-        constraint = add_cons_container!(container, cons_type, B, names, time_steps)
-        for t in time_steps, d in devices
-            min_rate = max(
-                PSY.get_active_power_limits_from(d).min,
-                PSY.get_active_power_limits_to(d).min,
-            )
-            max_rate = min(
-                PSY.get_active_power_limits_from(d).max,
-                PSY.get_active_power_limits_to(d).max,
-            )
-            constraint[PSY.get_name(d), t] = JuMP.@constraint(
-                container.JuMPmodel,
-                min_rate <= var[PSY.get_name(d), t] <= max_rate
-            )
-            # Needs refactoring. This add to expression model doesn't work anymore
-            # add_to_expression!(
-            #     container.expressions[ExpressionKey(ActivePowerBalance, PSY.Bus)],
-            #     PSY.get_number(PSY.get_arc(d).to),
-            #     t,
-            #     var[PSY.get_name(d), t],
-            #     -PSY.get_loss(d).l1,
-            #     -PSY.get_loss(d).l0,
-            # )
-        end
+
+    var = get_variable(container, FlowActivePowerVariable(), B)
+    constraint = add_cons_container!(container, cons_type(), B, names, time_steps)
+    for t in time_steps, d in devices
+        min_rate = max(
+            PSY.get_active_power_limits_from(d).min,
+            PSY.get_active_power_limits_to(d).min,
+        )
+        max_rate = min(
+            PSY.get_active_power_limits_from(d).max,
+            PSY.get_active_power_limits_to(d).max,
+        )
+        constraint[PSY.get_name(d), t] = JuMP.@constraint(
+            container.JuMPmodel,
+            min_rate <= var[PSY.get_name(d), t] <= max_rate
+        )
+        # Needs refactoring. This add to expression model doesn't work anymore
```

Comment:
To do this correctly you need to follow a pattern similar to the one in Services. @sourabhdalvi can probably help to get this done in the correct way.